addons/ofxGui/src/ofxInputField.cpp

```diff
@@ -788,6 +937,62 @@ bool ofxInputField<Type>::containsValidValue() const{
 	return validValue;
 }
 
+//-----------------------------------------------------------
+//   Simple button implementation
+//-----------------------------------------------------------
+template<typename Type>
+void ofxInputField<Type>::SimpleButton::setup(std::string name, bool bSetActive, const std::vector<unsigned char>& imgData){
+    this->name = name;
+    this->bActive = bSetActive;
+    if(imgData.size()){
+        this->img.allocate(17, 6, OF_IMAGE_COLOR_ALPHA);
+        ofxBaseGui::loadStencilFromHex(this->img, const_cast<unsigned char*>(&imgData[0]));
+        this->img.getTexture().setTextureMinMagFilter(GL_NEAREST,GL_NEAREST);
+    }
+}
+template<typename Type>
+void ofxInputField<Type>::SimpleButton::draw(){
+    ofPushStyle();
```

Comment:
HI. Sure. I understand how the rest of the gui works. maybe creating an fbo that renders the buttons in their different states, keep the texture, create a vbomesh for the buttons and then just bind the texture? it would be very much as how the text is handled. Would it make any difference-code wise and performance wise- to either store 3 textures and then bind each depending on the button state or have just one texture with the the 3 states on it and then change the uvcoords to adjust for each state of the button? I will test both options but if you already have an idea for which would be better please let me know. 
I will merge against master as well.

Reply:
loading the texture with an ofImage is fine i think the main problem is not that but that you are loading it per button. ofPush an pop style are super expensive as well. Of course having all the images in one texture would be better but there's no need for an fbo just encode the 6 images together and create a mesh with the corresponding texcture coordintates

Reply:
Yeah. I get it. I mean, having an fbo within a function, just for generating the textures, not to keep it, as these are created according to the font being used. I'll come back with a neat solution for it.
Cheers